file-ingestion-test/src/main/java/com/hazelcast/jet/tests/file/ingestion/FileIngestionTest.java

```diff
@@ -0,0 +1,244 @@
+    private static final int S3_CLIENT_CONNECTION_TIMEOUT_SECONDS = 10;
+    private static final int S3_CLIENT_SOCKET_TIMEOUT_MINUTES = 5;
+    private static final int DEFAULT_SLEEP_SECONDS = 4;
+    private static final int ITEM_COUNT = 1000;
+
+    private String hdfsUri;
+    private String hdfsPath;
+    private String bucketName;
+    private String s3Directory;
+    private String accessKey;
+    private String secretKey;
+    private int sleepSeconds;
+
+    public static void main(String[] args) throws Exception {
+        new FileIngestionTest().run(args);
+    }
+
+    @Override
+    protected void init(JetInstance client) throws Exception {
+        hdfsUri = property("hdfsUri", DEFAULT_HDFS_URI);
+        hdfsPath = hdfsUri + "/" + getClass().getSimpleName();
+
+        bucketName = property("bucketName", DEFAULT_BUCKET_NAME);
+        s3Directory = getClass().getSimpleName() + "/";
+        accessKey = property("accessKey", "");
+        secretKey = property("secretKey", "");
+
+        sleepSeconds = propertyInt("sleepSecondsBetweenJobs", DEFAULT_SLEEP_SECONDS);
+
+        createSourceFiles(client);
+    }
+
+    @Override
+    protected void test(JetInstance client, String name) {
+        long begin = System.currentTimeMillis();
+        int jobNumber = 0;
+        JobType[] jobTypes = JobType.values();
+        Job[] jobs = new Job[jobTypes.length];
+        while ((System.currentTimeMillis() - begin) < durationInMillis) {
+            for (int i = 0; i < jobTypes.length; i++) {
+                JobType jobType = jobTypes[i];
+                JobConfig jobConfig = new JobConfig();
+                jobConfig.setName(name + "-" + jobType + "-" + jobNumber);
+                jobs[i] = client.newJob(pipeline(jobType, jobNumber), jobConfig);
+            }
+            for (Job job : jobs) {
+                job.join();
+            }
+            for (JobType jobType : jobTypes) {
+                verifyObservable(client.getObservable(observableName(jobType, jobNumber)));
+            }
+            sleepSeconds(sleepSeconds);
+            jobNumber++;
+        }
+    }
+
+    private void verifyObservable(Observable<Long> observable) {
+        Iterator<Long> iterator = observable.iterator();
+        assertTrue(observable.name() + " is empty", iterator.hasNext());
+        assertEquals(ITEM_COUNT, iterator.next().intValue());
+        assertFalse(iterator.hasNext());
+        observable.destroy();
+    }
+
+    @Override
+    protected void teardown(Throwable t) {
+    }
+
+    private Pipeline pipeline(JobType jobType, int jobNumber) {
+        Pipeline p = Pipeline.create();
+
+        p.readFrom(source(jobType))
+                .groupingKey(ignored -> 0)
+                .aggregate(counting())
+                .map(Map.Entry::getValue)
+                .writeTo(Sinks.observable(observableName(jobType, jobNumber)));
+
+        return p;
+    }
+
+    private BatchSource<String> source(JobType jobType) {
```

Comment:
I think reading files is fundamental, we should cover avro,parquet,csv and json formats as well to make sure there are no memory leaks in the implementation.

Also the same code is also used for reading in SQL, even more important.